### PR TITLE
Archive Asset

### LINKED ISSUE #42: Convert Asset to Archived
As a User, I want to be able to convert an asset to archived once it has finished its life, so that more data for testing.
Acceptance Criteria:
- [x] UI Mock-up that fits our design
- [x] Implement UI
- [x] Button on "Asset Info" page that opens an interface
- [x] The interface has to be able to take in a timecycle as the "cycle of death"
- [x] Once you press "Save" it will delete all the sensor measurements that are over to the timecycle value
- [x] Set archived value of this object to true
- [ ] Unit Tests
- [x] Code convention (warnings in IDE, code cleanup)

### PR DESCRIPTION
**Pull request by JayChouteau**
This pull request covers issue #42.

IMPORTANT: Please note that the base branch of this pull-request is the revamp-UI. The reason is being that this builds on top of the work done in that pull request more than it does on top of the current develop branch. This will be changed once the revamp UI is merged into develop.

For this pull-request, a button was created following the new code conventions and revamp of UI for FXML, CSS, controller files. An archive button was created on the asset information page. That button opens a custom dialog that lets the user choose which time cycle should be considered as the last useful life cycle. When the selection is saved, the asset should not be visible in the assets thumbnails/list page anymore, but should still exist in the database (with the archived tag set to true). All measurements from after the selected time cycle should have been deleted in the database as well.

### CODE REVIEW

File: JavaFXUserInterface/src/main/java/Utilities/CustomDialog.java [hunk 1]
```diff
@@ -113,12 +94,81 @@ public static void addSystemControllerSaveDialog(MouseEvent mouseEvent) {
...
+     * @author Jeremie
+     */
+    public static void archiveAssetDialogShow(MouseEvent mouseEvent, Asset asset, Button parentSceneBtn) {
+        CustomDialog dialog = new CustomDialog(TextConstants.ARCHIVE_DIALOG_HEADER, TextConstants.ARCHIVE_DIALOG_CONTENT);
+        final Integer[] selectedCycle = new Integer[1];
+        AssetDAOImpl assetDAO = new AssetDAOImpl();
+
+        generateTimeCycleSelection(asset, dialog, selectedCycle);
+
+        dialog.getOkButton().setText("Save");
```
Comment (jeffbuonamici) [inline JavaFXUserInterface/src/main/java/Utilities/CustomDialog.java]: Make "Save" a constant

File: JavaFXUserInterface/src/main/java/Utilities/CustomDialog.java [hunk 2]
```diff
@@ -113,12 +94,81 @@ public static void addSystemControllerSaveDialog(MouseEvent mouseEvent) {
...
+     * functions allows to keep track of the selected time cycle.
+     *
+     * @param asset is the asset to be archived
+     * @param dialog is the parent custom dialog holding the label and ComboBox
+     * @param selectedCycle is the selected option in the ComboBox list
+     * @author Jeremie
+     */
+    private static void generateTimeCycleSelection(Asset asset, CustomDialog dialog, Integer[] selectedCycle) {
+        // Time Cycle Label Creation and Configuration
+        Label timeCycleLabel = new Label("Time Cycle:");
```
Comment (jeffbuonamici) [inline JavaFXUserInterface/src/main/java/Utilities/CustomDialog.java]: Make "Time Cycle:" a constant.

**Review by jeffbuonamici — CHANGES_REQUESTED** [review]

**Review by jeffbuonamici — APPROVED** [review]
Functionality of the issue works and my concerns were addressed.

**Review by Khal-Shah — APPROVED** [review]
The archive functionality is implemented well and meets the acceptance criteria. Selecting the last time cycle of asset's useful life is reflected properly both on the UI and the database. 

I feel the Cancel and Save buttons of the dialog box aren't aligned properly with the rest of the dialog box but that can be a bug for a future issue. 

Approved.

**Pull request merged**